torch/fx/passes/shape_prop.py

```diff
@@ -26,6 +26,7 @@ class TensorMetadata(NamedTuple):
     requires_grad: bool
     stride: Tuple[int, ...]
     memory_format: Optional[torch.memory_format]
+    device: torch.device
```

Comment:
Yeah, I think we should add this but probably in a saperate PR

Reply:
I separated the changes to the TensorMetadata out into a separate [PR](https://github.com/pytorch/pytorch/pull/142263)